drivers/usb/udc/udc_dwc2.c

```diff
@@ -65,6 +65,9 @@ enum dwc2_drv_event_type {
 /* Get Data FIFO access register */
 #define UDC_DWC2_EP_FIFO(base, idx)	((mem_addr_t)base + 0x1000 * (idx + 1))
 
+/* Percentage limit of how much SPRAM can be allocated for RxFIFO */
+#define MAX_RXFIFO_GDFIFO_PERCENTAGE 25
```

Comment:
When we introduced a maximum limit in percentage, maybe it could be better to make this constant configurable (I tried it with several values and seems that we can set the range `25-75`). WDYT ? 
 
Something like: 

```suggestion
#define MAX_RXFIFO_GDFIFO_PERCENTAGE CONFIG_UDC_DRIVER_RXFIFO_THRESHOLD
```

with the changes in `zephyr/drivers/usb/udc/Kconfig` :
```
config UDC_DRIVER_RXFIFO_USAGE_LIMIT
	int "RX FIFO usage limit (%)"
	range 25 75
	default 25
	help 
	  Limits how much of total available SPRAM (GDFIFOCFG field in GDFIFOCFG
	  register) can be used by RxFIFO.
```

Reply:
Why 25 to 75? If this ends up as Kconfig, then 100% should be the default (disable the limit altogether, which should remove the code that puts the limit in place) because the DWC2 Kconfig options are shared by all targets. You could then add different default based on your target.

Reply:
> then 100% should be the default 

Yes, agreed. 

May I ask is there any available Nordic board (dev kit) that can easily be used to verify usb applications?